wait-for-deployment

```diff
@@ -45,20 +45,21 @@ get_updated_replicas() {
   get_deployment_jsonpath '{.status.updatedReplicas}'
 }
 
-get_available_replicas() {
-  get_deployment_jsonpath '{.status.availableReplicas}'
+get_ready_replicas() {
+  get_deployment_jsonpath '{.status.readyReplicas}'
 }
 
 get_deployment_jsonpath() {
   local -r jsonpath="$1"
 
-  kubectl --namespace "${namespace}" get deployment "${deployment}" -o "jsonpath=${jsonpath}"
+  kubectl --namespace "${namespace}" get "${type}" "${deployment}" -o "jsonpath=${jsonpath}"
 }
 
 display_usage_and_exit() {
-  echo "Usage: $(basename "$0") [-n <namespace>] [-t <timeout>] <deployment>" >&2
+  echo "Usage: $(basename "$0") [-n <namespace>] [-t <timeout>] <deployment|statefulset> <resource-name>" >&2
```

Comment:
Instead of requiring another parameter that would break existing users, how about expecting callers to define the type as a slash-delimited prefix to the resource name like

```
[<resource type>/]<resource name>
```

where `<resource type>` is optional and defaults to `deployment`. This would also be in line with what you see in k8s land / kubectl.

Reply:
I like this idea. Unfortunately I don't have time to implement and test it in the moment, it will take a while.

Reply:
No hurry on my end (how could I expect that anyway ;) ), take your time!

Reply:
I've pushed a new commit, changing the script back to using one commandline argument only. This should do it :)